src/Uno.UI.Runtime.Skia.Wpf/Input/WpfKeyboardInputSource.cs

```diff
@@ -70,26 +72,32 @@ private void HostOnKeyUp(object sender, System.Windows.Input.KeyEventArgs args)
 				this.Log().Trace($"OnKeyPressEvent: {args.Key} -> {virtualKey}");
 			}
 
+			var scanCode = GetScanCode(args);
+
 			KeyUp?.Invoke(this, new(
 				"keyboard",
 				virtualKey,
 				GetKeyModifiers(args.KeyboardDevice.Modifiers),
 				new CorePhysicalKeyStatus
 				{
-					ScanCode = (uint)args.SystemKey,
+					ScanCode = scanCode,
 					RepeatCount = 1,
 				},
-				KeyCodeToUnicode((uint)args.SystemKey)));
+				KeyCodeToUnicode(scanCode)));
 		}
 		catch (Exception e)
 		{
 			Windows.UI.Xaml.Application.Current.RaiseRecoverableUnhandledException(e);
 		}
 	}
 
+	// WPF doesn't expose the scancode, but it exists internally
+	private static uint GetScanCode(KeyEventArgs args)
+		=> (uint)((int)(typeof(KeyEventArgs).GetProperty("ScanCode", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)?.GetValue(args) ?? 0));
```

Comment:
Thank you. Is it cheaper to use MapVirtualKey?

Reply:
I'm guessing a system call would be more expensive (at least theoretically, it would give the OS a reason to pre-empt the process), but the main reason is that I'm not sure the 2 approaches will always return the same thing. The docs say something about left/right-handedness and default values on failure, etc., so it's easier to just get what WPF gives us. We can re-evaluate later if this turns out to be a problem I guess.